src/server/game/Globals/ObjectMgr.cpp

```diff
@@ -762,6 +761,58 @@ void ObjectMgr::LoadCreatureTemplate(Field* fields, bool triggerHook)
 
 }
 
+void ObjectMgr::LoadCreatureTemplateModels()
+{
+    uint32 oldMSTime = getMSTime();
+
+    //                                               0           1                  2             3
+    QueryResult result = WorldDatabase.Query("SELECT CreatureID, CreatureDisplayID, DisplayScale, Probability FROM creature_template_model ORDER BY Idx ASC");
+
+    if (!result)
+    {
+        LOG_INFO("server.loading", ">> Loaded 0 creature template model definitions. DB table `creature_template_model` is empty.");
+        return;
+    }
+
+    uint32 count = 0;
+    do
+    {
+        Field* fields = result->Fetch();
+
+        uint32 creatureId = fields[0].Get<uint32>();
+        uint32 creatureDisplayId = fields[1].Get<uint32>();
+        float displayScale = fields[2].Get<float>();
+        float probability = fields[3].Get<float>();
```

Comment:
Somewhere we should check and log error if probability is >/< 0/1

Reply:
My concern with doing this is I know the sniffed data is not consistent with the probabilities, and I'm not sure if it's officially stored with a probability that could be considered invalid.

Reply:
okay but at least check if we are not less than 100%? Because that could be considered an issue?

Reply:
If I remember correctly (my memory ain't great), this implementation dynamically sizes the probability "table" for each display ID, so it can account for weird probability values that wouldn't add up to 100%.